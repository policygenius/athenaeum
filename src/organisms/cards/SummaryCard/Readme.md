### Examples:

```jsx
    <SummaryCard
      text={[
        {
          header: 'header',
          subheader: 'subheader'
        },
        {
          header: 'secondaryHeader',
          subheader: 'another subhead'
        }
      ]}
      icon='xIcon'
      onIconClick={() => alert('you clicked real good')}
      footerLinks={[
        {
          text: 'A link',
          onClick: () => alert('footer')
        },
        {
          text: 'second link',
          onClick: () => alert('second')
        }
      ]}
    />
```

With additional information as children
```jsx
    <SummaryCard
      text={[
        {
          header: 'header',
          subheader: 'subheader'
        },
      ]}
      icon='xIcon'
      onIconClick={() => alert('you clicked real good')}
      footerLinks={[
        {
          text: 'A link',
          onClick: () => alert('footer')
        },
        {
          text: 'second link',
          onClick: () => alert('second')
        }
      ]}
    >
      <Text size={9} font='b'>
        Additional information your can supply to the card.
      </Text>
    </SummaryCard>
```

```jsx
    <SummaryCard
      text={[
        {
          header: 'header',
          subheader: 'subheader'
        },
        {
          header: 'secondaryHeader',
          subheader: 'another subhead'
        }
      ]}
      icon='greenCheckWithCircle'
      footerLinks={[
        {
          text: 'A link',
          onClick: () => alert('footer')
        },
      ]}
    />
```
