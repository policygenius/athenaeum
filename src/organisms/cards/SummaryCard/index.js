import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import LinkWrapper from 'atoms/LinkWrapper';
import Text from 'atoms/Text';
import Spacer from 'atoms/Spacer';
import Icon from 'atoms/Icon';
import Layout from 'atoms/Layout';
import Col from 'atoms/Layout/Col';

import styles from './summary_card.module.scss';

function formatText(text) {
  let cols = [ 12 ];

  if (text.length > 1) {
    cols = [ 8, 4 ];
  }

  return (
    <Layout
      smallCols={cols}
    >
      {
        text.map((t, idx) =>
          <div
            key={`summary-card-lockup-${idx}`}
          >
            <Text size={5} font='a'>{t.header}</Text>
            <Spacer size={6} />
            <Text size={8} font='b' color='neutral-3'>{t.subheader}</Text>
          </div>
        )
      }
    </Layout>
  );
}

function formatFooter(links) {
  let cols = [ 12 ];

  if (links.length > 1) {
    cols = [ 8, 4 ];
  }

  return (
    <Layout
      smallCols={cols}
      className={styles['footer']}
    >
      {
        links.map((l, idx) =>
          <LinkWrapper
            type='secondary'
            onClick={l.onClick}
            key={`summary-card-footer-${idx}`}
          >
            <Text size={10} font='b' color='primary-3'>
              { l.text }
            </Text>
          </LinkWrapper>
        )
      }
    </Layout>
  );

}

function SummaryCard( props ) {
  const {
    text,
    footerLinks,
    icon,
    onIconClick,
    className,
    children
  } = props;

  const iconProps = {
    icon,
    className: styles['icon'],
  };

  if (onIconClick) {
    iconProps.onClick = onIconClick;
    iconProps.className = classnames(styles['icon'], styles['icon-click']);
  }

  return (
    <div className={classnames(styles['summary-card'], className)}>
      { icon && <Icon {...iconProps} /> }

      <div className={styles['content']}>
        { formatText(text) }

        {
            children &&
              <Col>
                <Spacer size={18} />
                { children }
              </Col>
          }
      </div>

      { footerLinks && formatFooter(footerLinks) }
    </div>
  );
}

SummaryCard.propTypes = {
  /**
   * class name added to class set
   */
  className: PropTypes.string,

  /**
   * Main header text of card
   */
  text: PropTypes.arrayOf(PropTypes.shape({
    header: PropTypes.string,
    subheader: PropTypes.string
  })).isRequired,

  /**
   * Optional footer with corresponding links
   */
  footerLinks: PropTypes.arrayOf(PropTypes.shape({
    text: PropTypes.string,
    onClick: PropTypes.func
  })),

  /**
   * Adds icon to upper right corner. See [Icon](#icon) for available values
   */
  icon: PropTypes.string,

  /**
   * Optional onClick function for icon
   */
  onIconClick: PropTypes.func
};

export default SummaryCard;
