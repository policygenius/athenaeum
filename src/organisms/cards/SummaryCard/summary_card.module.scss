@import '../shared/cards.module.scss';

.summary-card {
  @extend %card;
}

.content {
  padding: ru(.75) ru(.5);
}

.icon {
  @extend %icon;

  cursor: initial;
}

.icon-click {
  cursor: pointer;
}

.footer {
  @extend %footer;
  padding: ru(.5) ru(.5);
}
