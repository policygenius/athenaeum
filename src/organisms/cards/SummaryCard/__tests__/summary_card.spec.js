import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import SummaryCard from 'organisms/cards/SummaryCard';

describe('<SummaryCard />', () => {
  let props;

  beforeEach(() => {
    props = {
      text: [
        { header: 'header', subheader: 'subheader' }, { header: 'secondaryHeader', subheader: 'another subhead' }
      ],
      icon: 'xIcon',
      onIconClick: jest.fn(),
      footerLinks: [
        { text: 'A link', onClick: jest.fn() }, { text: 'second link', onClick: jest.fn() }
      ]
    };
  });

  it('renders correctly', () => {
    const wrapper = mount(<SummaryCard {...props} />);

    expect(wrapper.type()).toEqual(SummaryCard);
  });

  it('renders correctly', () => {
    const actual = renderer.create(<SummaryCard {...props} />).toJSON();

    expect(actual).toMatchSnapshot();
  });
});
