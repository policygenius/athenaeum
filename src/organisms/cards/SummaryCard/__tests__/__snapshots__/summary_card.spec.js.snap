// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryCard /> renders correctly 1`] = `
<div
  className="summary-card"
>
  <span
    className="icon-wrapper clickable icon icon-click"
    onClick={[MockFunction]}
    style={
      Object {
        "height": undefined,
        "width": undefined,
      }
    }
  >
    <img
      alt="xIcon"
      className="img"
      src="https://static.policygenius.com/svgs/xIcon.svgz"
      title="xIcon"
    />
  </span>
  <div
    className="content"
  >
    <div
      className="layout"
    >
      <div
        className="col col-sm-8 border-neutral-6"
      >
        <div>
          <p
            className="primary-3 type-a-5-bold inherit"
          >
            header
          </p>
          <div
            className="size-6"
          />
          <p
            className="neutral-3 type-b-8-medium inherit"
          >
            subheader
          </p>
        </div>
      </div>
      <div
        className="col col-sm-4 border-neutral-6"
      >
        <div>
          <p
            className="primary-3 type-a-5-bold inherit"
          >
            secondaryHeader
          </p>
          <div
            className="size-6"
          />
          <p
            className="neutral-3 type-b-8-medium inherit"
          >
            another subhead
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="layout footer"
  >
    <div
      className="col col-sm-8 border-neutral-6"
    >
      <a
        className="link neutral-1 secondary"
        onClick={[MockFunction]}
      >
        <p
          className="primary-3 type-b-10-medium inherit"
        >
          A link
        </p>
      </a>
    </div>
    <div
      className="col col-sm-4 border-neutral-6"
    >
      <a
        className="link neutral-1 secondary"
        onClick={[MockFunction]}
      >
        <p
          className="primary-3 type-b-10-medium inherit"
        >
          second link
        </p>
      </a>
    </div>
  </div>
</div>
`;
